src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] dataStorage;
+
+    public ArrayList() {
+        dataStorage = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        checkSizeForGrow();
+        dataStorage[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index != size) {
+            checkIndex(index);
+        }
+        checkSizeForGrow();
+        System.arraycopy(dataStorage, index, dataStorage, index + 1, size - index);
+        dataStorage[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list.size() > dataStorage.length - size) {
+            grow(size + list.size());
+        }
+        for (int i = 0; i < list.size(); i++) {
+            dataStorage[size++] = list.get(i);
+        }
```

Comment:
```suggestion
        for (int i = 0; i < list.size(); i++) {
            add(list.get(i));
        }
```
you can do like this